src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -420,15 +423,15 @@ static bool IsMessageExpired(Message receivedMessage, Dictionary<string, string>
                 return false;
             }
 
-            var sentDateTime = receivedMessage.GetSentDateTime(clockOffset);
-            var utcNow = DateTime.UtcNow;
+            var sentDateTime = receivedMessage.GetAdjustedDateTimeFromServerSetAttributes("SentTimestamp", clockOffset);
             var expiresAt = sentDateTime + timeToBeReceived;
-            if (expiresAt > utcNow)
+            if (expiresAt > DateTime.UtcNow)
             {
                 return false;
             }
+
             // Message has expired.
-            Logger.Info($"Discarding expired message with Id {messageId}, expired {utcNow - expiresAt} ago at {expiresAt} utc.");
+            Logger.Info($"Discarding expired message with Id {messageId}, expired {DateTime.UtcNow - expiresAt} ago at {expiresAt} utc.");
```

Comment:
Should we be storing `Now` above and using it in both places?

Reply:
Good point @mikeminutillo, fixed.